apps/E2E/src/CheckboxLegacy/pages/CheckboxLegacyPageObject.ts

```diff
@@ -5,61 +5,52 @@ import {
+import { Attribute, AttributeValue } from '../../common/consts';
 
 class CheckboxLegacyPageObject extends BasePage {
   /******************************************************************/
   /**************** UI Element Interaction Methods ******************/
   /******************************************************************/
   async isCheckboxChecked(): Promise<boolean> {
-    return await (await this._primaryComponent).isSelected();
+    const checkbox = await this._primaryComponent;
+    if (this.platform === DesktopPlatform.Windows) {
+      // for native windows, .isSelected() always returns false. this is a workaround
+      return (await checkbox.getAttribute(Attribute.ToggleState)) === AttributeValue.on;
+    } else {
+      return await checkbox.isSelected();
+    }
   }
 
-  async waitForCheckboxChecked(timeout?: number): Promise<void> {
-    await browser.waitUntil(async () => await this.isCheckboxChecked(), {
-      timeout: timeout ?? this.waitForUiEvent,
-      timeoutMsg: 'The Checkbox was not toggled correctly',
-      interval: 1000,
-    });
+  /* Waits for the checkbox to be checked or unchecked if the new state is true or false. Returns true if the
+   * checkbox toggled to the new state. */
+  async waitForCheckboxToggle(newState: boolean, errorMessage?: string): Promise<boolean> {
+    if ((await this.isCheckboxChecked()) !== newState) {
+      await this.waitForCondition(async () => (await this.isCheckboxChecked()) === newState, errorMessage);
+    }
+    return (await this.isCheckboxChecked()) === newState;
   }
 
-  /* Useful in beforeEach() hook to reset the checkbox before every test */
-  async toggleCheckboxToUnchecked(): Promise<void> {
-    if (await this.isCheckboxChecked()) {
-      await (await this._primaryComponent).click();
+  /* Toggles the checkbox to the checked if newState == true or unchecked if newState == false. */
+  async toggleCheckbox(newState: boolean) {
+    if ((await this.isCheckboxChecked()) !== newState) {
+      await this.click(this._primaryComponent);
+      await this.waitForCheckboxToggle(
+        newState,
+        `Clicked the primary checkbox to turn it ${newState ? 'on' : 'off'}, but it failed to change states.`,
+      );
     }
   }
 
-  async didOnChangeCallbackFire(): Promise<boolean> {
-    const callbackText = await By(CHECKBOX_ON_PRESS);
-    await browser.waitUntil(async () => await callbackText.isDisplayed(), {
-      timeout: this.waitForUiEvent,
-      timeoutMsg: 'The OnChange callback did not fire.',
-      interval: 1000,
-    });
-
+  /* Unlike snapshot testing, we cannot spy on functions to determine if they get called or not. In order to determine if
+   * the onChange() callback gets fired, we show / hide the a Text label as the callback gets fired. This way, we know that
+   * the onChange() callback has fired by checking that the label element is currently displayed. */
+  async didOnChangeCallbackFire(errorMsg?: string): Promise<boolean> {
```

Comment:
Same as above

Reply:
Turns out i forgot to commit this 🤦. No longer optional now.